8347270: Remove unix_getParentPidAndTimings, unix_getChildren and unix_getCmdlineAndUserInfo

MBaesken opened the pull request:
The functions unix_getParentPidAndTimings and unix_getCmdlineAndUserInfo in src/java.base/unix/native/libjava/ProcessHandleImpl_unix.c used to hold an implementation that was shared between Solaris and AIX. Linux and MacOS already had specific implementations. After the removal of the Solaris port, these two functions can be removed in favor of inline implementations of os_getParentPidAndTimings and os_getCmdlineAndUserInfo in src/java.base/aix/native/libjava/ProcessHandleImpl_aix.c.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8347270](https://bugs.openjdk.org/browse/JDK-8347270): Remove unix_getParentPidAndTimings, unix_getChildren and unix_getCmdlineAndUserInfo (**Bug** - P4)


### Reviewers
 * [Joachim Kern](https://openjdk.org/census#jkern) (@JoKern65 - Committer) Review applies to [d49d3d36](https://git.openjdk.org/jdk/pull/23013/files/d49d3d3616fac93730cf2f9095723f33ae083193)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/23013/head:pull/23013` \
`$ git checkout pull/23013`

Update a local copy of the PR: \
`$ git checkout pull/23013` \
`$ git pull https://git.openjdk.org/jdk.git pull/23013/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 23013`

View PR using the GUI difftool: \
`$ git pr show -t 23013`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/23013.diff">https://git.openjdk.org/jdk/pull/23013.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/23013#issuecomment-2580805395)
</details>

JoKern65 reviewed (approved):
LGTM

RogerRiggs reviewed (approved):
Nice cleaup.  Thanks

RealCLanger reviewed (changes requested):
Looks good, but I have a few requests/suggestions:

1. Please update the copyright years
2. In ProcessHandleImpl_aix.c you could add an SAP copyright line
3. In line 57 of src/java.base/unix/native/libjava/ProcessHandleImpl_unix.c there is another mentioning of Solaris which could be removed
4. On the same lines as you could get rid of unix_getChildren in src/java.base/unix/native/libjava/ProcessHandleImpl_unix.c and move the implementation to os_getChildren in src/java.base/linux/native/libjava/ProcessHandleImpl_linux.c

MBaesken commented:
Hi Christoph, I adjusted the COPYRIGHT info and removed unix_getChildren .
Regarding 'Solaris' mentioning in this file , I removed that one too but a grep shows there are plenty of  Solaris comments etc. in the codebase.

RealCLanger commented on src/java.base/unix/native/libjava/ProcessHandleImpl_unix.c:
```diff
@@ -78,7 +78,7 @@
  *   ProcessHandleImpl_<os>.c file.
  *
- * Example 1:
+ * Example :
```
```suggestion
 * Example:
```

MBaesken commented:
Thanks for the reviews !

/integrate
